src/DSi_TMD.h

```diff
@@ -0,0 +1,157 @@
+/*
+    Copyright 2016-2023 melonDS team
+
+    This file is part of melonDS.
+
+    melonDS is free software: you can redistribute it and/or modify it under
+    the terms of the GNU General Public License as published by the Free
+    Software Foundation, either version 3 of the License, or (at your option)
+    any later version.
+
+    melonDS is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+    FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with melonDS. If not, see http://www.gnu.org/licenses/.
+*/
+
+#ifndef DSI_TMD_H
+#define DSI_TMD_H
+
+#include "types.h"
+#include <array>
+
+namespace DSi_TMD
+{
+
+struct [[gnu::packed]] TitleMetadataContent {
+    /// Content ID (00,00,00,vv) ;lowercase/hex ;"0000000vv.app"
+    u8 ContentId[4];
+
+    /// Content Index (00,00)
+    u16 ContentIndex;
```

Comment:
Do we need to comment each and every one of these? Some comments aren't really adding much info.